### ISSUE TITLE
crash in fit_OSLLifeTimes()

### ISSUE THREAD

**Issue by mcol** [R-Lum/Luminescence#182]
```R
fit_OSLLifeTimes(ExampleData.TR_OSL, signal_range = 1:2)
# Error in fit_OSLLifeTimes(ExampleData.TR_OSL, signal_range = 1:2) : 
#  object 'start_parameters' not found
```

**Comment by mcol** [comment]
The problem here is that the signal range contains only 2 points, and when we get here:
https://github.com/R-Lum/Luminescence/blob/759a1142651cc557f9fd06c6b061d1aff244cdf9/R/fit_OSLLifeTimes.R#L432-L433
we don't even enter the `while` loop (as `df2 = length(df[[2]]) - 2 * m - 2` is negative: `2 - 2 * 1 - 2 = -2`), and `start_parameters` doesn't get initialised.

It seems that we need a test on a minimum number of signal points (which depends on the requested number of components `m`).

**Comment by mcol** [comment]
Actually we have a check, but that happens before we restrict the dataset to only the points in the signal range requested, so it should be enough to move that to the correct spot.

**Comment by mcol** [comment]
Fixed by #195.
